x-pack/plugins/observability_solution/inventory/server/routes/entity_definition/get_entity_definitions.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import * as t from 'io-ts';
+import { createInventoryServerRoute } from '../create_inventory_server_route';
+
+export const getEntityDefinitionSourceIndexPatternsByType = createInventoryServerRoute({
+  endpoint: 'GET /internal/inventory/entity/definitions/sources',
+  params: t.type({
+    query: t.type({
+      type: t.string,
+    }),
+  }),
+  options: {
+    tags: ['access:inventory'],
+  },
+  async handler({ params, request, plugins }) {
+    const entityManagerStart = await plugins.entityManager.start();
+    const { type } = params.query;
+    const entityManagerClient = await entityManagerStart.getScopedClient({ request });
+
+    const entityDefinitionsSource = await entityManagerClient.v2.readSourceDefinitions({ type });
+
+    return {
+      definitionIndexPatterns: entityDefinitionsSource.map(
```

Comment:
related to the above comment, should this be a flat array of every source index_patterns ?